src/gui/gui_console.cpp

```diff
@@ -1191,7 +1191,7 @@ void ui::console_edit::edit_box_enter(sys::state& state, std::string_view s) noe
 		command::c_complete_constructions(state, state.local_player_nation);
 		break;
 	case command_info::type::instant_research:
-		command::c_change_research_points(state, state.local_player_nation, float(640000000.f));
+		state.user_settings.instant_research_enabled = !state.user_settings.instant_research_enabled;
```

Comment:
This would also be a problem for multiplayer: any user-initiated change to the game state that can affect how the game progresses must be wrapped in a command so that it can be sent to the server which then synchronizes it with any other commands to produce a single authoritative sequence of the order in which things happen and what happened